UpsertSpecialServiceActivationActivity/__tests__/handler.test.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { toCosmosErrorResponse } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import { context } from "../../__mocks__/durable-functions";
+import { aFiscalCode, cgnActivatedDates } from "../../__mocks__/mock";
+import {
+  CardPending,
+  StatusEnum
+} from "../../generated/definitions/CardPending";
+import {
+  CardRevoked,
+  StatusEnum as RevokedStatusEnum
+} from "../../generated/definitions/CardRevoked";
+import { UserCgn } from "../../models/user_cgn";
+import {
+  ActivityInput,
+  getUpsertSpecialServiceActivationActivityHandler
+} from "../handler";
+import { Activation } from "../../generated/services-api/Activation";
+import { ServiceId } from "../../generated/services-api/ServiceId";
+import { ActivationStatusEnum } from "../../generated/services-api/ActivationStatus";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+
+const aServiceId = "SERVICE_ID" as ServiceId;
+const aSpecialServiceActivation: Activation = {
+  fiscal_code: aFiscalCode,
+  service_id: aServiceId,
+  status: ActivationStatusEnum.ACTIVE,
+  version: 1 as NonNegativeInteger
+};
+const upsertServiceActivationMock = jest
+  .fn()
+  .mockImplementation(() =>
+    TE.of({ status: 200, value: aSpecialServiceActivation })()
+  );
+
+const servicesClientMock = {
+  upsertServiceActivation: upsertServiceActivationMock
+} as any;
+
+const aNotFoundResponse = { status: 404, value: "Not found" };
+
+const anActivityInput: ActivityInput = {
+  activationStatus: ActivationStatusEnum.ACTIVE,
+  fiscalCode: aFiscalCode
+};
+describe("UpsertSpecialServiceActivationActivity", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it("should return failure if an error occurs during input decode retrieve", async () => {
+    const upsertSpecialServiceActivationActivity = getUpsertSpecialServiceActivationActivityHandler(
+      servicesClientMock
+    );
+    const response = await upsertSpecialServiceActivationActivity(context, {
+      aaa: "wrongActivityInput"
+    });
+    expect(response.kind).toBe("FAILURE");
+  });
+
+  it("should return failure if an error occurs during special service activation's upsert call", async () => {
+    upsertServiceActivationMock.mockImplementationOnce(() =>
+      Promise.reject("Connectivity Error")
+    );
+    const upsertSpecialServiceActivationActivity = getUpsertSpecialServiceActivationActivityHandler(
+      servicesClientMock
+    );
+    const response = await upsertSpecialServiceActivationActivity(
+      context,
+      anActivityInput
+    );
+    expect(response.kind).toBe("FAILURE");
+    if (response.kind === "FAILURE") {
+      expect(response.reason).toBe("Connectivity Error");
+    }
+  });
```

Comment:
This type of error should return an exception, not an Activity permanent failure. (related to the comment inside the `handler.ts`)